runtime/signals.c

```diff
@@ -254,6 +254,9 @@ void caml_request_major_slice (int global)
   }else{
     Caml_state->requested_major_slice = 1;
   }
+  /* Beginning the mark phase requires emptying the minor heap */
```

Comment:
Why does the beginning of the mark phase require an empty heap? What will go wrong if it is not?